lightning/src/ln/channel.rs

```diff
@@ -790,752 +873,275 @@ pub(super) struct Channel<Signer: ChannelSigner> {
-			funding_tx_confirmation_height: 0,
-			short_channel_id: None,
-			channel_creation_height: current_chain_height,
-
-			feerate_per_kw: msg.feerate_per_kw,
-			channel_value_satoshis: msg.funding_satoshis,
-			counterparty_dust_limit_satoshis: msg.dust_limit_satoshis,
-			holder_dust_limit_satoshis: MIN_CHAN_DUST_LIMIT_SATOSHIS,
-			counterparty_max_htlc_value_in_flight_msat: cmp::min(msg.max_htlc_value_in_flight_msat, msg.funding_satoshis * 1000),
-			holder_max_htlc_value_in_flight_msat: Self::get_holder_max_htlc_value_in_flight_msat(msg.funding_satoshis, &config.channel_handshake_config),
-			counterparty_selected_channel_reserve_satoshis: Some(msg.channel_reserve_satoshis),
-			holder_selected_channel_reserve_satoshis,
-			counterparty_htlc_minimum_msat: msg.htlc_minimum_msat,
-			holder_htlc_minimum_msat: if config.channel_handshake_config.our_htlc_minimum_msat == 0 { 1 } else { config.channel_handshake_config.our_htlc_minimum_msat },
-			counterparty_max_accepted_htlcs: msg.max_accepted_htlcs,
-			holder_max_accepted_htlcs: cmp::min(config.channel_handshake_config.our_max_accepted_htlcs, MAX_HTLCS),
-			minimum_depth: Some(cmp::max(config.channel_handshake_config.minimum_depth, 1)),
-
-			counterparty_forwarding_info: None,
-
-			channel_transaction_parameters: ChannelTransactionParameters {
-				holder_pubkeys: pubkeys,
-				holder_selected_contest_delay: config.channel_handshake_config.our_to_self_delay,
-				is_outbound_from_holder: false,
-				counterparty_parameters: Some(CounterpartyChannelTransactionParameters {
-					selected_contest_delay: msg.to_self_delay,
-					pubkeys: counterparty_pubkeys,
-				}),
-				funding_outpoint: None,
-				opt_anchors: if opt_anchors { Some(()) } else { None },
-				opt_non_zero_fee_anchors: None
-			},
-			funding_transaction: None,
-
-			counterparty_cur_commitment_point: Some(msg.first_per_commitment_point),
-			counterparty_prev_commitment_point: None,
-			counterparty_node_id,
-
-			counterparty_shutdown_scriptpubkey,
-
-			commitment_secrets: CounterpartyCommitmentSecrets::new(),
-
-			channel_update_status: ChannelUpdateStatus::Enabled,
-			closing_signed_in_flight: false,
-
-			announcement_sigs: None,
-
-			#[cfg(any(test, fuzzing))]
-			next_local_commitment_tx_fee_info_cached: Mutex::new(None),
-			#[cfg(any(test, fuzzing))]
-			next_remote_commitment_tx_fee_info_cached: Mutex::new(None),
-
-			workaround_lnd_bug_4006: None,
-			sent_message_awaiting_response: None,
-
-			latest_inbound_scid_alias: None,
-			outbound_scid_alias,
-
-			channel_pending_event_emitted: false,
-			channel_ready_event_emitted: false,
-
-			#[cfg(any(test, fuzzing))]
-			historical_inbound_htlc_fulfills: HashSet::new(),
-
-			channel_type,
-			channel_keys_id,
-
-			pending_monitor_updates: Vec::new(),
-		};
-
-		Ok(chan)
+	/// Returns true if funding_created was sent/received.
+	pub fn is_funding_initiated(&self) -> bool {
+		self.channel_state >= ChannelState::FundingSent as u32
 	}
```

Comment:
IIUC, `OutboundV1Channel` and `InboundV1Channel` can neither be in this state. Thus, it's fine that we ignore the corresponding maps when writing `ChannelManager`. As of now `Channel` may or not be in this state, though. Going forward, what should be represented as a `ChannelState` variant vs defined by a specific channel struct type? I think preferably we don't have redundant state to avoid inconsistency.

Reply:
This would make sense just to have in `Channel`. I'll leave it as is for now and address it in a further "remove redundant state" PR